app/source/js/viewmodels/ServiceProfessionalService.js

```diff
@@ -80,7 +84,7 @@ function ServiceProfessionalServiceViewModel(app) {
             });
         }
         
-        if (!this.isSelectionMode()) {
+        if (!this.isSelectionMode() || !this.loadEmptyPricingTypes()) {
```

Comment:
@situatedbit This block adds the empty groups to the list, so, as I understand the meaning of the *loadEmptyPricingTypes* property, it must check for the positive value of that property not for the negative as here.
I tried the three activities involved:
- A. List service professional services serviceProfessionalService/:jobTitleID (on my test /serviceProfessionalService/106)
- B. Create service professional booking /appointment/-4
- C. Create client booking /booking/:userID/:jobTitleID  (on my test /booking/userID/jobTitleID)

With a massage therapist profile that has packages and services but no add-ons, results are:
- A. OK. Displays the empty add-on group at the end with just the 'add new' option.
- B. KO. Does not display the empty add-on group at all, and must do it just to have the 'add new' option.
- C. KO. It displays the add-on group, just the header since has no items.

Aside from that bug, a proposal: I'm thinking in to drop the check of *isSelectionMode* in order to depend only on *loadEmptyPricingTypes*, make it explicit rather than semi-automatic. We would set the value on each use/activity for the wanted result. Makes sense to you? .. or just drop the exclamation mark :-)